src/interfaces/coral_web/src/components/ShareModal.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState } from 'react';
+
+import { Button, Icon, Input, Spinner, Text } from '@/components/Shared';
+import { env } from '@/env.mjs';
+import { useCreateSnapshotLinkId, useSnapshots } from '@/hooks/snapshots';
+
+type ShareModalProps = {
+  conversationId: string;
+};
+
+export const ShareModal: React.FC<ShareModalProps> = ({ conversationId }) => {
+  const [status, setStatus] = useState<
+    'modal-loading' | 'modal-error' | 'update-url-loading' | 'update-url-error' | undefined
+  >('modal-loading');
+  const { loadingSnapshots, getSnapshotLinksByConversationId, deleteAllSnapshotLinks } =
+    useSnapshots();
+  const { mutateAsync: createSnapshotLinkId } = useCreateSnapshotLinkId();
+  const snapshotLinks = getSnapshotLinksByConversationId(conversationId);
+  const snapshotLinksExists = !!snapshotLinks
+    ? snapshotLinks && snapshotLinks.length > 0
+    : undefined;
+
+  const latestSnapshotLink = snapshotLinks[snapshotLinks.length - 1];
+  const [linkId, setLinkId] = useState(latestSnapshotLink ?? '');
+
+  useEffect(() => {
+    if (loadingSnapshots) return;
+
+    if (latestSnapshotLink && latestSnapshotLink) {
+      setLinkId(latestSnapshotLink);
+      setStatus(undefined);
+    } else {
+      const generateNewSnapshotLink = async () => {
+        setStatus('modal-loading');
+        try {
+          const linkId = await createSnapshotLinkId({ conversationId });
+          if (linkId) {
+            setLinkId(linkId);
+            setStatus(undefined);
+          } else {
+            setStatus('modal-error');
```

Comment:
when does this happen?

Reply:
ah good catch-- copied this from coral where the `linkId` could be null but that's not the case here